gammapy/utils/random.py

```diff
@@ -39,38 +40,32 @@ def sample_sphere(size, lon_range=None, lat_range=None, unit='rad'):
     ----------
     size : int
         Number of samples to generate
-    lon_range : tuple
-        Longitude range (min, max) tuple in range (0, 360)
-    lat_range : tuple
-        Latitude range (min, max) tuple in range (-90, 90)
-    units : {'rad', 'deg'}
-        Units of input range and returned angles
+    lon_range : `~astropy.units.Quantity`, optional
+        Longitude range (min, max) in range (0, 360) deg
+    lat_range : `~astropy.units.Quantity`, optional
+        Latitude range (min, max) in range (-90, 90) deg
 
     Returns
     -------
-    lon, lat: arrays
+    lon, lat: `~astropy.units.Quantity`
         Longitude and latitude coordinate arrays
     """
-    # Correctly interpret units at an early stage
-    if unit in ['rad', 'radian', 'radians']:
-        unit = 'rad'
-    elif unit in ['deg', 'degree', 'degrees']:
-        unit = 'deg'
-    else:
-        raise ValueError('Invalid unit: {0}'.format(unit))
-
     # Convert inputs to internal format (all radians)
     size = int(size)
 
-    if (lon_range is not None) and (unit == 'deg'):
-        lon_range = np.radians(lon_range)
+    if lon_range is not None:
+        lon_unit = lon_range.unit
+        lon_range.to('radian')
```

Comment:
@mapazarr – Here I'm not sure what the best API is.
Is it better to always return angles in the same units as the inputs?
What's more intuitive / convenient for callers?